platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -341,18 +341,18 @@ mender_storage_get_artifact_name(char **artifact_name) {
 }
 
 mender_err_t
-mender_storage_save_update_state(mender_update_state_t state, const char *artifact_type) {
+mender_storage_save_update_state(const mender_update_state_t state, const char *artifact_type) {
     assert(NULL != artifact_type);
 
     size_t artifact_type_len;
 
-    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, state, sizeof(state))) {
+    if (!checked_nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, (const void *)state, sizeof(state))) {
```

Comment:
Should it be `void *` or `unsigned char *`?

Reply:
The `nvs_write` API call takes `void *`. We could cast to char, but then it will anyway be interpreted as void... so I think is best to do it at once here.